crates/slot-arithmetic/src/lib.rs

```diff
@@ -15,7 +15,60 @@
 use minicbor::{Decode, Decoder, Encode};
 use serde::{Deserialize, Serialize};
 
-type Slot = u64;
+#[derive(Clone, Debug, Copy, PartialEq, PartialOrd, Eq, Serialize, Deserialize)]
+#[repr(transparent)]
+pub struct Slot(u64);
+
+#[derive(Debug, PartialEq, Eq, thiserror::Error)]
+pub enum SlotArithmeticError {
+    #[error("slot arithmetic overflow, substracting {1} from {0}")]
+    Underflow(u64, u64),
+}
+
+impl Slot {
+    pub const fn new(slot: u64) -> Self {
```

Comment:
Looks like `Slot(1)`, `Slot::new(1)` and `From::from(1)` are equivalent? Maybe we could align on `Slot(1)` that looks more idiomatic for those cases then?
I would expect most of those to disappear once we introduce the new `Slot` in most places.

Reply:
Ah, I forgot to delete `new`. `Slot(1)` is only possible locally as the "field" is private, so one must use `From`